internal/icinga2/client_api.go

```diff
@@ -306,17 +318,38 @@ func (client *Client) checkMissedChanges(ctx context.Context, objType string, ca
 		var fakeEv *event.Event
 		if checkableIsMuted && attrs.Acknowledgement != AcknowledgementNone {
 			ackComment, err := client.fetchAcknowledgementComment(ctx, hostName, serviceName, attrs.AcknowledgementLastChange.Time())
-			if err != nil {
-				return fmt.Errorf("fetching acknowledgement comment for %q failed, %w", objectName, err)
-			}
+			if errors.Is(err, errMissingAcknowledgementComment) {
+				// Unfortunately, there is no Acknowledgement object in Icinga 2, but only related runtime attributes
+				// attached to Host or Service objects. Those attributes contain no authorship. The only way to link an
+				// acknowledgement to a contact, when being fetched through the Config Objects API, is to find a
+				// matching Comment object, which contains an author field.
+				//
+				// This is not the case for the Event Stream API, where AcknowledgementSet has an author field.
+				//
+				// However, when no author is present, the Acknowledgement Event cannot be processed. Eventually, the
+				// Incident.processAcknowledgementEvent method will fail hard.
+
+				client.Logger.Infow("Cannot find the comment for an acknowledgement, creating a generic muted event",
+					zap.String("object", objectName), zap.Error(err))
+
+				fakeEv, err = client.buildCommonEvent(ctx, hostName, serviceName)
+				if err != nil {
+					return fmt.Errorf("failed to construct checkable fake unmute event: %w", err)
+				}
 
-			ack := &Acknowledgement{Host: hostName, Service: serviceName, Author: ackComment.Author, Comment: ackComment.Text}
-			// We do not need to fake ACK set events as they are handled correctly by an incident and any
-			// redundant/successive ACK set events are discarded accordingly.
-			ack.EventType = typeAcknowledgementSet
-			fakeEv, err = client.buildAcknowledgementEvent(ctx, ack)
-			if err != nil {
-				return fmt.Errorf("failed to construct Event from Acknowledgement response, %w", err)
+				fakeEv.Type = event.TypeMute
+				fakeEv.SetMute(false, "Acknowledgement event without corresponding comment")
```

Comment:
Why are you generating a fake `mute` event without setting its `mute` flag to true? This event type combination is considered to be invalid by the way, but since we do not validate all the events from the Icinga 2 package, you got passed through without an error, otherwise you would have run into this problem.
https://github.com/Icinga/icinga-notifications/blob/388732218bcaf20b07876c5d2fc2fe865f3857b0/internal/event/event.go#L92-L94

I would also use something like this as a mute reason: `Checkable is acknowledged, but we could not find its corresponding comment`